reefperf/testing_tool_wrapper.py

```diff
@@ -10,16 +10,59 @@ class TestingToolWrapper(object, metaclass=ABCMeta):
     def run_tool(self, node, app):
         pass
 
+    @property
+    @abstractmethod
+    def supported_resources(self):
+        pass
+
+
+class NotFoundSupportedResource(Exception):
+    pass
+
 
 testing_tools = ClassRegistry()
 
 
 @testing_tools.register("SleepWrapper")
-class SleepWrapper(object):
+class SleepWrapper(TestingToolWrapper):
     def __init__(self, sleep_length):
         self._sleep_length = sleep_length
 
+    @property
+    def supported_resources(self):
+        return ()
+
     def run_tool(self, node, app):
         # TODO: sleep on cloud node
         time.sleep(self._sleep_length)
         return f"Slept for {self._sleep_length} seconds"
+
+
+@testing_tools.register("WrkWrapper")
+class WrkWrapper(TestingToolWrapper):
+    def __init__(self, threads, connections, duration):
+        self._threads = threads
+        self._connections = connections
+        self._duration = duration
+
+    def set_up(self, test_node):
+        test_node.connection.exec_command("sudo apt-get update && sudo apt-get -y install wrk")
+
+    @property
+    def supported_resources(self):
+        return 'https', 'http'
+
+    def get_resource_location(self, app):
```

Comment:
this method gets the location of the **first** resource, if available. I suggest to call it `get_first_available_resource_location`